projects/js-packages/components/components/threats-data-views/index.tsx

```diff
@@ -328,7 +354,8 @@ export default function ThreatsDataViews( {
 						},
 				  ]
 				: [] ),
-			...( dataFields.includes( 'firstDetected' )
+			...( dataFields.includes( 'firstDetected' ) &&
+			( statusFilters.includes( 'fixed' ) || statusFilters.includes( 'ignored' ) )
```

Comment:
I am curious if there's a possible alternative configuration that would allow us to still include the "first detected" field regardless of the current filters, but hide/show the particular field in the table view based on the current filters, instead of omitting the definition of the fields entirely. I'll follow up on this!

Reply:
>I am curious if there's a possible alternative configuration that would allow us to still include the "first detected" field regardless of the current filters, but hide/show the particular field in the table view based on the current filters, instead of omitting the definition of the fields entirely.

I had a similar hope, and did come across `[isVisible](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#isvisible)` which I thought _might_ be an option but seemed to not function as I expected it to.

Reply:
Could we use [isVisible](https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/types.ts#L129)?

Reply:
I am working on a potential approach using the `onChangeView` callback, will share shortly!

Reply:
Just a heads up that I did start with the onChangeView approach and it seemed to have its own limitations - IIRC there were challenges related to not being able to check the current view state against the changed view state because it was always outdated. Regardless, interested to see what you think might work reintroducing that take.